woocommerce/woocommerce-ios
# [Mobile Payments] Display battery too low to connect message when battery too low to connect

Closes: #5986
Closes: #5993

@joshheald or @ctarda - just one review is needed but either of y'all are welcome

@adamzelinski - this adds a new user experience - let me know if it is OK

### Description
- Adds a new modal similar to the "bad postal code" modal that can appear when attempting to connect to a reader
- Adds it for when the Stripe Terminal SDK (2.5 or newer) informs us (during connection) that the reader battery is too low to connect to

### Testing instructions
- Revert this commit to force the error to happen: https://github.com/woocommerce/woocommerce-ios/commit/a7ba93ad87cc4763e04f6b83e37022ba3f826b45
- Start the connection flow (Menu > Settings > In-Person Payments > Manage Card Reader)
- Attempt to connect to the reader
- Observe the error
- Use the retry button to try again, or the cancel button to stop
- Ensure new unit test passes

### Screenshots

<img src="https://user-images.githubusercontent.com/1595739/151448163-c9ff12fd-08e2-4b9a-83ea-16fa1240f16d.PNG" width=60%/>

---
